sklearn/tests/test_docstring_parameters.py

```diff
@@ -232,37 +223,35 @@ def test_fit_docstring_attributes(name, Estimator):
         "FeatureUnion",
     ):
         est = _construct_compose_pipeline_instance(Estimator)
-    elif Estimator.__name__ in (
-        "SparseCoder",
-        "GaussianRandomProjection",
-        "SparseRandomProjection",
-    ):
-        est = _construct_projection(Estimator)
+    elif Estimator.__name__ == "SparseCoder":
+        est = _construct_sparse_coder(Estimator)
     else:
         est = _construct_instance(Estimator)
 
     if Estimator.__name__ == 'SelectKBest':
-        est.k = 2
-
-    if Estimator.__name__ == 'DummyClassifier':
-        est.strategy = "stratified"
-
-    if 'PLS' in Estimator.__name__ or 'CCA' in Estimator.__name__:
-        est.n_components = 1  # default = 2 is invalid for single target.
+        est.set_params(k=2)
+    elif Estimator.__name__ == 'DummyClassifier':
+        est.set_params(strategy="stratified")
+    elif Estimator.__name__ in ('CCA', 'PLS'):
+        # default = 2 is invalid for single target.
+        est.set_params(n_components=1)
```

Comment:
this new formulation is not equivalent to the previous one. I think you should revert

Reply:
I changed it :) The tests were failing so they work :)

Reply:
mb didn't see you already changed it